worlds/oot_soh/location_access/dungeons/dodongos_cavern.py

```diff
@@ -4,219 +4,406 @@
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOMB_ROOM_UPPER.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_FAR_BRIDGE.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_FAR_BRIDGE.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOMB_ROOM_UPPER.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_FAR_BRIDGE.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_LOBBY.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_REGION.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_LOBBY.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_REGION.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BACK_ROOM.value),
-        rule=lambda state: can_break_mud_walls(state, world))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_REGION.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOSS_REGION.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOSS_ROOM.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BOSS_ROOM.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY.value))
-
-    world.get_region(Regions.DODONGOS_CAVERN_BACK_ROOM.value).connect(
-        world.get_region(Regions.DODONGOS_CAVERN_BOSS_REGION.value))
-
-
-def set_location_rules(world: "SohWorld") -> None:
-    player = world.player
-
-    set_rule(world.get_location(Locations.DODONGOS_CAVERN_MAP_CHEST.value),
-             rule=lambda state: can_break_mud_walls(state, world)
-             or state.has(Items.STRENGTH_UPGRADE.value, player))
-
-    set_rule(world.get_location(Locations.DODONGOS_CAVERN_END_OF_BRIDGE_CHEST.value),
-             rule=lambda state: can_break_mud_walls(state, world))
-
-    set_rule(world.get_location("Dodongos Cavern Lower Lizalfos"),
-             rule=lambda state: can_kill_enemy(state, world, Enemies.LIZALFOS, EnemyDistance.CLOSE, quantity=2))
+def set_region_rules(world: "SohWorld") -> None:
+    # Dodongos Cavern Entryway
+    connect_regions(Regions.DODONGOS_CAVERN_ENTRYWAY, world, [
+        (Regions.DODONGOS_CAVERN_BEGINNING, lambda bundle: True),
+        (Regions.DEATH_MOUNTAIN_TRAIL, lambda bundle: True),
+    ])
+
+    ## Dodongos Cavern Beginning
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_BEGINNING, world, [
+        (Regions.DODONGOS_CAVERN_ENTRYWAY, lambda bundle: True),
+        (Regions.DODONGOS_CAVERN_LOBBY,
+         lambda bundle: blast_or_smash(bundle) or has_item(Items.STRENGTH_UPGRADE, bundle))
+    ]),
+
+    ## Dodongos Cavern Lobby
+    # Events
+    add_events(Regions.DODONGOS_CAVERN_LOBBY, world, [
+        (EventLocations.DODONGOS_CAVERN_GOSSIP_STONE_LOBBY, Events.CAN_ACCESS_FAIRIES, lambda bundle: True),
```

Comment:
```suggestion
        (EventLocations.DODONGOS_CAVERN_GOSSIP_STONE_LOBBY, Events.CAN_ACCESS_FAIRIES, lambda bundle: (can_break_mud_walls(bundle) or has_item(Items.GORONS_BRACELET, bundle)) and call_gossip_fairy(bundle)),
```
Pretty sure this is what this is supposed to be. Please double check.